pkg/target/http_templating_test.go

```diff
@@ -73,6 +73,34 @@ func TestHTTP_Templating_NoPrettyPrinting(t *testing.T) {
 	assert.Empty(invalidMessages)
 }
 
+func TestHTTP_Templating_WithEnvVariable(t *testing.T) {
+	assert := assert.New(t)
+
+	t.Setenv("API_KEY", "supersecret")
+
+	rawTemplate :=
+		`{
+  "api_key": "{{ env "API_KEY" }}",
+  "events": [{{range $i, $data := .}}{{if $i}},{{end}}{{prettyPrint .}}{{end}}]
+}`
+
+	parsedTemplate, err := parseRequestTemplate(rawTemplate)
+	assert.Nil(err)
+	target := HTTPTarget{requestTemplate: parsedTemplate}
+
+	inputMessages := []*models.Message{
+		{Data: []byte(`{ "data": 1}`)},
+		{Data: []byte(`{ "data": 2}`)},
+	}
+
+	templated, goodMessages, invalidMessages := target.renderBatchUsingTemplate(inputMessages)
+
+	expectedOutput := "{\n  \"api_key\": \"supersecret\",\n  \"events\": [{\"data\":1},{\"data\":2}]\n}"
```

Comment:
[FYI](https://pkg.go.dev/github.com/stretchr/testify/assert#JSONEq)

Reply:
Nice, exactly what I needed!! Changed a bit formatting in tests then to make it more readable